server/src/internalClusterTest/java/org/opensearch/action/admin/indices/rollover/RolloverIT.java

```diff
@@ -263,23 +262,21 @@ public void testRolloverDryRun() throws Exception {
         ensureGreen();
         Logger allocationServiceLogger = LogManager.getLogger(AllocationService.class);
 
-        MockLogAppender appender = new MockLogAppender();
-        appender.start();
-        appender.addExpectation(
-            new MockLogAppender.UnseenEventExpectation(
-                "no related message logged on dry run",
-                AllocationService.class.getName(),
-                Level.INFO,
-                "*test_index*"
-            )
-        );
-        Loggers.addAppender(allocationServiceLogger, appender);
+        final RolloverResponse response;
+        try (MockLogAppender appender = MockLogAppender.createForLoggers(allocationServiceLogger)) {
+            appender.addExpectation(
+                new MockLogAppender.UnseenEventExpectation(
+                    "no related message logged on dry run",
+                    AllocationService.class.getName(),
+                    Level.INFO,
+                    "*test_index*"
+                )
+            );
 
-        final RolloverResponse response = client().admin().indices().prepareRolloverIndex("test_alias").dryRun(true).get();
```

Comment:
nit - do we need to pull out this declaration?

Reply:
I just wanted to keep the same logic of making the assertions after the appender was stopped and removed so moving the declaration out of the try-with-resources block was necessary. I could move the assertions inside this block to avoid moving the declaration if you prefer that.